File: AD-HOC/1129/main.js
let input = require('fs').readFileSync('/dev/stdin', 'utf-8');
let lines = input.split('\n');

let alternatives = ['A', 'B', 'C', 'D', 'E'];
let i = 0;
let n = parseInt(lines[i++]);
while (n !== 0) {
    for (let j = 0; j < n; j++) {
        let choice = lines[i++].split(' ').flatMap((mean, i) => parseInt(mean) <= 127 ? alternatives[i] : []);
        if (choice.length == 1) {
            console.log(choice[0]);
        } else {
            console.log('*');
    }
    } n = parseInt(lines[i++]);
}

File: AD-HOC/1129/main.py
alternatives = ['A', 'B', 'C', 'D', 'E']
while (n := int(input())):
    for _ in range(n):
        choice = [alternatives[i] for i, mean in enumerate(map(int, input().split())) if mean <= 127]
        if len(choice) == 1:
            print(choice[0])
        else:
            print('*')